### DISCUSSION TITLE
Integration von Sungrow PV-Invertern

### DISCUSSION THREAD

**Discussion by DB1113** [fhempy/fhempy#114] [Ideas]
Hallo,

super Projekt, für mich eines der wichtigsten innerhalb fhem!
Nochmals vielen Dank dafür.

Noch eine Idee:
Die Einbindung von Sungrow PV-Invertern über ModbusTCP.
Zu den Invertern gibt es schon eine Diskussion im fhem-Forum: [Einbindung eines Sungrow SH10RT: SH05,SH06,SH08,SH10](https://forum.fhem.de/index.php/topic,115597.0.html)
Hier ist es aber so, dass jedes Attribut natürlich wieder angelegt werden muss.

Ich habe schon mal auf Github geschaut... und hier gibt es auch schon ein paar Projekte, die sich mit den Sungrow-Geräten beschäftigen. Es muss also nicht alles neu erfunden werden.
Hier Links: 
[ModbusTCP2MQTT)](https://github.com/MatterVN/ModbusTCP2MQTT)
[Sungrow-SHx-Inverter-Modbus-Home-Assistant](https://github.com/mkaiser/Sungrow-SHx-Inverter-Modbus-Home-Assistant)
[SungrowInverter](https://github.com/mvandersteen/SungrowInverter)

Und als Kirsche auf dem Sahnehäubchen: der Upload auf pvOutput! 🥇 Link: [https://pvoutput.org/](https://pvoutput.org/)
Dazu habe ich einen Code gefunden: [sungrow2pvoutput](https://github.com/kronicd/sungrow2pvoutput)

Vielleicht als Anregung, falls es langweilig geworden sein sollte über die Feiertage... :D
Könnte man diese Geräte in fhempy mit integrieren?

Danke für eine kurze Rückmeldung.

Bis denn! :)

**Comment by fhempy** [comment]
Definitiv interessant und eine Integration sollte auch einfach möglich sein mit dieser Library:
https://github.com/mvandersteen/SungrowInverter

Im Moment fehlt mir noch etwas die Zeit um was neues zu integrieren. Ich möchte vorher noch ein paar Bugs fixen :)

Aber gerne auch mal selbst ausprobieren. Mit ein bisschen Programmierkenntnissen schafft man das schon.

**Comment by DB1113** [comment]
Hi,
Vielen Dank für die Rückmeldung.
Bei dem Code, den du genannt hast, können dem Inverter aber keine Befehle gegeben werden, oder?

Mit meinen "Programmmierkenntnissen" kann ich da nicht helfen. Sorry.
Für mich ist dann warten angesagt... :) 

Hab nur festgestellt, dass in "helloworld" nicht soo viel drin ist... ;)
EDIT: Doch da is was drin... Hab ich jetzt gefunden^^

Bis bald.
